### ISSUE TITLE
should i decrease the value of the gt when i compute the "predict_flow"?

### ISSUE THREAD

**Issue by Lvhhhh** [lmb-freiburg/flownet2#56]
i am back again! should i decrease the value of the gt when i compute the "predict_flow" because the size of the flow_gt is smaller than the input image . the value of gt is related to the size of the input pictures~

**Comment by nikolausmayer** [comment]
Hi,
sorry, I'm not sure what you mean. Are you asking about our training files? Or do you have your own training data where the image resolution is different from the flow resolution?

**Comment by Lvhhhh** [comment]
i use the dispnet to predict the disparity. the disparity means the 'X' coordinate differences between the left images and the right images.  when the size of the left and right images reduce the double . the value of it should reduce the double too. so i want to know whether the value of the disparity gt changes?

**Comment by nikolausmayer** [comment]
Ok, let's look at an example:

You have two images, *left* and *right*. Their resolution is e.g. *512x384* pixels. The pixel at (x,y) position (200,100) in the left image is found at (150,100) in the right image. That pixel's correct disparity is *50*.

Now you upsample both images to double resolution, i.e. *1024x768*. Upscaling images linearly scales all spatial relations. This means that the pixel is now located at (400,200) in the left image and (300,200) in the right image. Disparity is also a spatial relationship, so it scales linearly as well. The pixel's correct new disparity is *100*.

If you upsample your images for training, you must upsample the groundtruth and linearly scale its values.

**Comment by Lvhhhh** [comment]
yeah!! what i say is: in training phase. the net output six predict_disp. they are different sizes and they mean the prediction of different size pictures. the disparity ground truth of them are all "Downsample" by the ground truth of disparity. i want to know whether to change their values of each ground truth more than change their values??

**Comment by nikolausmayer** [comment]
Ah, I get it. No, the intermediate (lower-resolution) outputs do not have to be correctly scaled disparity maps. I'm pretty sure we just nearest-neighbor downsampled the data without rescaling. At those points we do not need technically correct outputs, and the difference is just a linear factor which the network could easily learn anyway.

**Comment by Lvhhhh** [comment]
i know it. whether correct the outputs or not have little effect to the network. it is just a linear factor. 
now , i found the training does not converge. and i will add a type of loss which l2 loss between img0 and warpped image(image warpped from img1) at each intermediate(lower-resolution) outputs. of course. the img0 ,img1 and their predict disparity are rescaled according to their size. Now, i found it is necessary to change disparity values according to their sizes to makes the disparity mean the absolute coordinate differences between img0 and img1.

**Comment by eddy-ilg** [comment]
Dear Lvhhh, 

we kindly ask you not to post questions in the form "I typed this command and hit enter - and it didn't work". Please do only ask questions when you have understood the problem thoroughly and provide a detailed description. We weill not answer questions where you don't take the time to formulate a clear inquiery and then expect us to spend large amounts of time to figure this out. 

Regards, 

Eddy Ilg
